src/cpp/core/include/core/r_util/RActiveSessionsStorage.hpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * RActiveSessionsStorage.hpp
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#ifndef CORE_R_UTIL_R_ACTIVE_SESSIONS_STORAGE
+#define CORE_R_UTIL_R_ACTIVE_SESSIONS_STORAGE
+
+
+#include <core/r_util/RActiveSession.hpp>
+#include <shared_core/Error.hpp>
+
+namespace rstudio {
+namespace core {
+namespace r_util {
+
+   class IActiveSessionsStorage
+   {
+      public:
+         virtual core::Error createSession(const std::string& id, std::map<std::string, std::string> initialProperties) = 0;
+         virtual std::vector< boost::shared_ptr<ActiveSession> > listSessions(FilePath userHomePath, bool projectSharingEnabled) const = 0;
```

Comment:
Just talking about this with Andrew - I think it would make sense to have the `IActiveSessionsStorage` only return the list of session IDs and have the `ActiveSessions` object to the validation it wants to do.